### PR TITLE
chore(eslint-plugin)!: remove dependencies from ESLint plugin

### LINKED ISSUE #10740: Remove third-party configurations for `eslint-plugin-patternfly-react`
The `eslint-plugin-patternfly-react` package contains linting presets for [various ESLint plugins](https://github.com/patternfly/patternfly-react/blob/1f81cd766361955fc6776b8e67c5fe60ecc8a372/packages/eslint-plugin-patternfly-react/package.json#L25-L38), these are all listed as explicit dependencies. This is an issue, as anyone who installs the ESLint plugin will now also get all these other ESLint plugins installed that they likely do not need, and if they do are quite possibly a version that conflicts.

Instead of having a preset for all these third-party plugins the ESLint plugin should be stripped so that it only contains it's own rules and presets, and remove the dependency on any third-party ESLint plugin. This will reduce complexity and maintenance burden, and pull it more in-line with how users expect ESLint plugins to work.

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #10740

BREAKING CHANGE: this removes the bundled dependencies and pre-defined rules from the ESLint 'reccomended' configuration.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     "eslint-plugin-markdown": "^3.0.1",
     "eslint-plugin-prettier": "^5.1.3",
     "eslint-plugin-react": "^7.34.2",
+    "eslint-plugin-react-hooks": "^4.6.2",
```
Comment (jonkoops) [inline package.json:52]: Had to move this plugin to the root, as it is used by the linting configuration of the project. It was previously included incidentally as a dependency of the ESLint plugin.

File: package.json [hunk 2]
```diff
@@ -46,9 +46,11 @@
     "babel-jest": "^29.7.0",
     "concurrently": "^8.2.2",
     "eslint": "^9.7.0",
+    "eslint-config-prettier": "^9.1.0",
     "eslint-plugin-markdown": "^5.1.0",
-    "eslint-plugin-prettier": "^5.1.3",
+    "eslint-plugin-prettier": "^5.2.1",
     "eslint-plugin-react": "^7.34.2",
+    "eslint-plugin-react-hooks": "^4.6.2",
```
Comment (jonkoops) [inline package.json:53]: Had to move some dependencies to the root, as it is used by the linting configuration of the project. It was previously included incidentally as a dependency of the ESLint plugin.

File: packages/eslint-plugin-patternfly-react/README.md [hunk 3]
```diff
@@ -34,17 +34,3 @@ module.exports = {
...
-yarn build
-```
-
-Note the build scripts for this are located in the root package.json under `yarn build`.
-
-### Publishing
-
-```
-yarn publish
-```
```
Comment (jonkoops) [inline packages/eslint-plugin-patternfly-react/README.md:50]: These instructions do not apply to this package, they were likely copied from a template before, so I removed them here.

**Review by kmcfaul — APPROVED** [review]
We may want to double check that we don't need to set up any additional lint plugins / rule overrides in our config after these get removed.

**Comment by jonkoops** [comment]
> We may want to double check that we don't need to set up any additional lint plugins / rule overrides in our config after these get removed.

I think this should not be an issue as the recommended rules from the plugin are not actually used by the configuration.

**Review by tlabaj — APPROVED** [review]

**Pull request merged**